### PR TITLE
144 update vue2 to vue3 in visualisation website

### LINKED ISSUE #144: Update Vue2 to Vue3 in visualisation website
~~I will have to work on this one since it has inter dependencies with other projects that need to be carefully unwound~~

Dependencies are successfully unwound and the migration is ready to be started.
I think the Vue version will need to be update in both this repo and geo-visualisation-components

### LINKED ISSUE #205: Automatically close linked issues when the pull request is merged
This functionality works when merging into the master branch but not when merging into the otakaro branch.

### PR DESCRIPTION
**Pull request by LukeParky**
DESCRIPTION OF PR:
Migrates the Vue app from Vue 2 to Vue 3. Doing so allows better security, new features, better support.
Migrates from options API to composition API.

Also changed from Webpack build tool to reccomeneded Vite build tool, decreasing build times and complexity significantly.
Closes #144 
Closes #205 - This PR is not directly related to #205 but by landing the PR we can use it to test #205.

## Developer Checklist
- [X] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [ ] Ensure these tests have the expected behaviour
  - [ ] Test locally and ensure tests are passing
- [X] Update documentation
  - [X] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

File: visualisation/entrypoint.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@
 # Used by docker to import environment variables into a docker container without baking them into the image
 
 ROOT_DIR=/app
-# Find the list of VUE_APP_ variables in the environment variables
-VUE_ENV_VARS=$(printenv | grep -o "VUE_APP_\w*")
+# Find the list of VITE_ variables in the environment variables
+VUE_ENV_VARS=$(printenv | grep -o "VITE_\w*")
 
 echo "Replacing env constants in JS"
 echo "Env constants = ${VUE_ENV_VARS}"
```
Comment (LukeParky) [inline visualisation/entrypoint.sh:10]: This should be VITE_ENV_VARS

File: visualisation/index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <link rel="icon" href="/favicon.ico">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Flood Resilience Digital Twin (FReDT)</title>
```
Comment (LukeParky) [inline visualisation/index.html:7]: Test this works with title mixin

File: visualisation/src/mixins/title.ts [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+//   /**
+//    * Changes the document title prefix to the page title
+//    */
+//   created: function (this: App): void {
+//     const pageTitle = this.$options.title;
+//     if (pageTitle) {
+//       document.title = `${pageTitle} | ${appBaseTitle}`;
+//     }
+//   }
+// }; todo reinstate this functionality
```
Comment (LukeParky) [inline visualisation/src/mixins/title.ts:19]: todo

File: visualisation/vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export default defineConfig({
+  plugins: [
+    vue(),
+    viteStaticCopy({
+      targets: [
+        {src: './node_modules/cesium/Build/Cesium/Assets', dest: './'},
+        {src: './node_modules/cesium/Build/Cesium/ThirdParty', dest: './'},
+        {src: './node_modules/cesium/Build/Cesium/Widgets', dest: './'},
+        {src: './node_modules/cesium/Build/Cesium/Workers', dest: './'},
+      ]
```
Comment (LukeParky) [inline visualisation/vite.config.ts:16]: Add comment

**Comment by LukeParky** [comment]
Investigating potential bug around using environment variables in build

**Comment by LukeParky** [comment]
Investigating bug related to memory management in the front end

File: visualisation/package.json [hunk 5]
```diff
@@ -2,40 +2,40 @@
...
   "scripts": {
-    "serve": "vue-cli-service serve",
-    "build": "vue-cli-service build",
-    "lint": "eslint src --ext .js,.ts,.vue"
+    "serve": "vite",
+    "build": "run-p type-check \"build-only {@}\" --",
+    "preview": "vite preview",
+    "build-only": "vite build",
+    "type-check": "vue-tsc --build --force",
+    "lint": "eslint src --ext .js,.ts,.vue --fix --ignore-path .gitignore",
```
Comment (LukeParky) [inline visualisation/package.json:12]: Ensure warnings are shown as errors in linting, ensure fixes are pushed by actions, and ensure type-check is performed by action

File: visualisation/README.md [hunk 6]
```diff
@@ -25,4 +25,4 @@
    ```bash
    npm run serve
    ```
-1. Access website: [localhost:8080](https://localhost:8080)
+1. Access website: [localhost:5173](https://localhost:8080)
```
Comment (caseyli92) [inline visualisation/README.md:28]:
```suggestion
1. Access website: [localhost:5173](https://localhost:5173)
```

**Review by caseyli92 — APPROVED** [review]
I noticed one thing that might need correction, but other than that, everything looks good to me. Since I don’t have front-end experience, I don’t have much feedback to offer. Just be sure to test everything thoroughly to ensure it works as expected.

**Pull request merged**